ui/src/data/reducers/index.ts

```diff
@@ -0,0 +1,35 @@
+// Libraries
+import {produce} from 'immer'
+
+// Actions
+import {Action} from 'src/data/actions'
+
+export interface DataState {
+  queryResultsByQueryID: {[queryID: string]: string[]}
+}
+
+export const initialState: DataState = {
+  queryResultsByQueryID: {},
+}
+
+export const dataReducer = (
+  state: DataState = initialState,
+  action: Action
+): DataState => {
+  switch (action.type) {
+    case 'SET_QUERY_RESULTS_BY_QUERY': {
+      return produce(state, draftState => {
+        const {queryID, files} = action
+        if (queryID && files.length) {
```

Comment:
I think we should be checking these conditions before firing `SET_QUERY_RESULTS_BY_QUERY` action, and then we can rely on them being here.

Reply:
Sounds good.